solver/llbsolver/ops/merge.go

```diff
@@ -70,6 +70,9 @@ func (m *mergeOp) Exec(ctx context.Context, g session.Group, inputs []solver.Res
 		if !ok {
 			return nil, errors.Errorf("invalid reference for merge %T", inp.Sys())
 		}
+		if wref.ImmutableRef == nil {
```

Comment:
Isn't this an error?

Reply:
This case happens if you do something contrived like `Merge(Diff(Image("busybox"), Image("busybox")), foo)`, i.e. you do a diff where upper+lower end up being the same ref and thus simplifying to `nil`, which then gets plugged into merge. Basically, a nil input is treated as `Scratch` here.

It would be nicer if we handle all cases of `Diff(a,b)` where `a == b` on the client side, but I'm not sure it's possible in general, for instance in cases where `a` and `b` are images w/ different image refs that resolve to the same digest at the time of the solve.